File: webgl/src/js/ViewToneMapping.js
// ViewToneMapping.js

var GL = bongiovi.GL;
var gl;
var glslify = require("glslify");

function ViewToneMapping() {
	this.exposure = 2.5;
	bongiovi.ViewCopy.call(this, null, glslify("../shaders/toneMapping.frag"));
}

var p = ViewToneMapping.prototype = new bongiovi.ViewCopy();
p.constructor = ViewToneMapping;

p.render = function(texture) {
	this.shader.bind();
	this.shader.uniform("texture", "uniform1i", 0);
	this.shader.uniform("exposure", "uniform1f", this.exposure);
	texture.bind(0);
	GL.draw(this.mesh);
};

module.exports = ViewToneMapping;

File: cinder/Entrainment/src/ViewToneMapping.cpp
//
//  ViewToneMapping.cpp
//  Entrainment03
//
//  Created by Yi-Wen Lin on 2015/7/28.
//
//

#include "ViewToneMapping.h"
#include "GlobalSettings.h"
#include "MeshUtils.h"

using namespace ci;
using namespace ci::app;
using namespace std;

ViewTonMapping::ViewTonMapping() : ViewCopy("shaders/copy.vert", "shaders/toneMapping.frag") {
}

ViewTonMapping::ViewTonMapping(string vsPath, string fsPath) : ViewCopy(vsPath, fsPath) {
}

void ViewTonMapping::render(gl::TextureRef texture) {
    render(*texture);
}

void ViewTonMapping::render(gl::Texture texture) {
    shader->bind();
    shader->uniform("texture", 0);
    shader->uniform("exposure", GlobalSettings::getInstance().exposure);
    texture.bind();
    gl::draw(mesh);
    texture.unbind();
    shader->unbind();
}